### PR TITLE
Don't throw exception

### LINKED ISSUE #484: Clicking settings button leads to error
Hi!

We are using ownCloud 10.3.1 on Ubuntu 18.04.3 LTS and Apache 2.4.29. We are able to login with our LDAP accounts but clicking on the settings button produces the following error:

image
image

```
Dec  5 09:52:54 HOSTNAME ownCloud[13271]: [JE5FttI4uTGhBEru5Jda][IP][89591EE5-532F-4A06-BBDE-E0762256B3DE][index][GET][/owncloud/settings/personal] Exception: {"Exception":"OCA\User_LDAP\Exceptions\DoesNotExistOnLDAPException","Message":"89591EE5-532F-4A06-BBDE-E0762256B3DE","Code":0,"Trace":"
#0 /var/www/owncloud/lib/private/User/User.php(361): OCA\User_LDAP\User_Proxy->canChangeAvatar('89591EE5-532F-4...')
#1 /var/www/owncloud/settings/Panels/Personal/Profile.php(101): OC\User\User->canChangeAvatar()
#2 /var/www/owncloud/settings/Controller/SettingsPageController.php(167): OC\Settings\Panels\Personal\Profile->getPanel()
#3 /var/www/owncloud/settings/Controller/SettingsPageController.php(85): OC\Settings\Controller\SettingsPageController->getPanelsData(Array)
#4 /var/www/owncloud/lib/private/AppFramework/Http/Dispatcher.php(153): OC\Settings\Controller\SettingsPageController->getPersonal('general')
#5 /var/www/owncloud/lib/private/AppFramework/Http/Dispatcher.php(85): OC\AppFramework\Http\Dispatcher->executeController(Object(OC\Settings\Controller\SettingsPageController), 'getPersonal')
#6 /var/www/owncloud/lib/private/AppFramework/App.php(100): OC\AppFramework\Http\Dispatcher->dispatch(Object(OC\Settings\Controller\SettingsPageController), 'getPersonal')
#7 /var/www/owncloud/lib/private/AppFramework/Routing/RouteActionHandler.php(47): OC\AppFramework\App::main('SettingsPageCon...', 'getPersonal', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
#8 /var/www/owncloud/lib/private/Route/Router.php(342): OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
#9 /var/www/owncloud/lib/base.php(920): OC\Route\Router->match('/settings/perso...')
#10 /var/www/owncloud/index.php(54): OC::handleRequest()
#11 {main}","File":"/var/www/owncloud/apps/user_ldap/lib/User_Proxy.php","Line":244}
```

This error does not occur if i use an internal user.

If i comment the line 244 in "*\owncloud\apps\user_ldap\lib\User_Proxy.php" the error does not occur and i am able to enter "settings".

Maybe that's a problem with our ldap, can anyone point me to the right direction?

Regards

### PR DESCRIPTION
**Pull request by IljaN**
Fixes #484

### CODE REVIEW

**Review by karakayasemi — COMMENTED** [review]
IMHO, catching all the possible exceptions in a try block and returning false on a catch block would be better.

**Comment by IljaN** [comment]
@karakayasemi Done

File: lib/User_Proxy.php [hunk 1]
```diff
@@ -236,19 +236,19 @@ public function getDisplayName($uid) {
...
-		foreach ($this->backends as $backend) {
-			if ($backend->userExists($uid)) {
-				return $backend->canChangeAvatar($uid);
+		try {
+			foreach ($this->backends as $backend) {
+				if ($backend->userExists($uid)) {
+					return $backend->canChangeAvatar($uid);
+				}
 			}
+		} catch (\Exception $e) {
```
Comment (jvillafanez) [inline lib/User_Proxy.php:247]: Is it worthy to log something, maybe for tracing purposes? Catching 4-5 exceptions (according to what has been deleted) seem to be a bunch of things that could go wrong, so we might want to know what happened. Maybe a debug message with the exception message is enough, we don't need to log the exception if we don't want to make too much noise (we're returning false already).

**Comment by jvillafanez** [comment]
Just a little thing. Code looks good :+1:

**Review by jvillafanez — APPROVED** [review]

**Pull request merged**